easybuild/tools/utilities.py

```diff
@@ -300,22 +300,21 @@ def time2str(delta):
     if not isinstance(delta, datetime.timedelta):
         raise EasyBuildError("Incorrect value type provided to time2str, should be datetime.timedelta: %s", type(delta))
 
-    delta_secs = delta.days * 3600 * 24 + delta.seconds + delta.microseconds / 10**6
-
-    if delta_secs < 60:
-        res = '%d sec' % int(delta_secs)
-    elif delta_secs < 3600:
-        mins = int(delta_secs / 60)
-        secs = int(delta_secs - (mins * 60))
-        res = '%d min %d sec' % (mins, secs)
-    else:
-        hours = int(delta_secs / 3600)
-        mins = int((delta_secs - hours * 3600) / 60)
-        secs = int(delta_secs - (hours * 3600) - (mins * 60))
-        hours_str = 'hours' if hours > 1 else 'hour'
-        res = '%d %s %d min %d sec' % (hours, hours_str, mins, secs)
+    delta_secs = delta.total_seconds()
 
-    return res
+    hours = int(delta_secs / 3600)
+    delta_secs -= hours * 3600
+    mins = int(delta_secs / 60)
+    secs = int(delta_secs - mins * 60)
+
+    res = []
+    if hours:
+        res.append('%d %s' % (hours, 'hour' if hours == 1 else 'hours'))
+    if mins or hours:
+        res.append('%d %s' % (mins, 'min' if mins == 1 else 'mins'))
+    res.append('%d %s' % (secs, 'sec' if secs == 1 else 'secs'))
```

Comment:
I suspect this is going to break some users' grep scripts :)

Reply:
Well, I hope they don't grep for the "secs" part, but for "FINISHED IN" (or what that was) ;)